Desciption: WiFi.cpp triggers GCC bug, this is a workaround until gcc bug is fixed
Author: Scott Howard <showard@debian.org>
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=698254
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/arduino/+bug/1112499
Bug-Upstream: http://gcc.gnu.org/bugzilla/show_bug.cgi?id=50925

Index: arduino/hardware/arduino/avr/platform.txt
===================================================================
--- arduino.orig/hardware/arduino/avr/platform.txt	2014-03-07 12:45:02.892146502 -0500
+++ arduino/hardware/arduino/avr/platform.txt	2014-03-07 12:45:31.816147189 -0500
@@ -14,12 +14,12 @@
 # Default "compiler.path" is correct, change only if you want to overidde the initial value
 #compiler.path={ide.path}/tools/avr/bin/..
 compiler.c.cmd=avr-gcc
-compiler.c.flags=-c -g -Os -w -ffunction-sections -fdata-sections -MMD -D__PROG_TYPES_COMPAT__
+compiler.c.flags=-c -g -Os -w -ffunction-sections -fdata-sections -MMD -D__PROG_TYPES_COMPAT__ -fno-caller-saves
 compiler.c.elf.flags=-Os -Wl,--gc-sections
 compiler.c.elf.cmd=avr-gcc
 compiler.S.flags=-c -g -x assembler-with-cpp
 compiler.cpp.cmd=avr-g++
-compiler.cpp.flags=-c -g -Os -w -fno-exceptions -ffunction-sections -fdata-sections -MMD -D__PROG_TYPES_COMPAT__
+compiler.cpp.flags=-c -g -Os -w -fno-exceptions -ffunction-sections -fdata-sections -MMD -D__PROG_TYPES_COMPAT__ -fno-caller-saves
 compiler.ar.cmd=avr-ar
 compiler.ar.flags=rcs
 compiler.objcopy.cmd=avr-objcopy
